tortoise/queryset.py

```diff
@@ -184,7 +184,10 @@ def resolve_ordering(
                 annotation_info = annotation.resolve(self.model, table)
                 self.query = self.query.orderby(annotation_info["field"], order=ordering[1])
             else:
-                field_object = self.model._meta.fields_map.get(field_name)
+                if self.model != model:
```

Comment:
Would this not mean that when self.model == model the `model` var is still the correct one?
Just do the `field_object = model._meta.fields_map.get(field_name)` instead.

Reply:
I tried to just replace self.model with model but didn't seems to work

Reply:
I found that just change model is okay with TCs. How about remove unnecessary logic

``` diff
diff --git a/tortoise/queryset.py b/tortoise/queryset.py
index d09ce89..68f6647 100644
--- a/tortoise/queryset.py
+++ b/tortoise/queryset.py
@@ -184,13 +184,9 @@ class AwaitableQuery(Generic[MODEL]):
                 annotation_info = annotation.resolve(self.model, table)
                 self.query = self.query.orderby(annotation_info["field"], order=ordering[1])
             else:
-                if self.model != model:
-                    field_object = model._meta.fields_map.get(field_name)
-                else:
-                    field_object = self.model._meta.fields_map.get(field_name)
-
+                field_object = model._meta.fields_map.get(field_name)
                 if not field_object:
-                    raise FieldError(f"Unknown field {field_name} for model {self.model.__name__}")
+                    raise FieldError(f"Unknown field {field_name} for model {model.__name__}")
                 field_name = field_object.source_field or field_name
                 field = getattr(table, field_name)

```